sdk/core/azure-core-test/src/main/java/com/azure/core/test/TestBase.java

```diff
@@ -131,18 +147,65 @@ protected void beforeTest() {
     protected void afterTest() {
     }
 
+    /**
+     * Get test arguments need to run for the test framework based on the service version.
+     *
+     * @param serviceVersionList The service version argument for the parameterized tests.
+     * @param serviceVersionEnv Indicates whether the service version parameters need to rolling.
+     * @return Stream of arguments for parameterized test framework.
+     */
+    public static Stream<Arguments> getArgumentsFromServiceVersion(List<ServiceVersion> serviceVersionList,
+        String serviceVersionEnv) {
+        if (!ARGUMENTS_LIST.isEmpty()) {
+            return ARGUMENTS_LIST.stream();
+        }
+        int serviceVersionCount = serviceVersionList.size();
+        List<HttpClient> httpClientList = getHttpClients();
+        int httpClientCount = httpClientList.size();
+        boolean rollingHttpClient = HTTP_CLIENT_FROM_ENV.equalsIgnoreCase(AZURE_TEST_HTTP_CLIENTS_VALUE_ROLLING);
+        boolean rollingServiceVersion = serviceVersionEnv != null
+            && serviceVersionEnv.equalsIgnoreCase(AZURE_TEST_SERVICE_VERSIONS_VALUE_ROLLING);
+        for (ServiceVersion s: serviceVersionList) {
+            for (HttpClient h: httpClientList) {
+                ARGUMENTS_LIST.add(Arguments.of(h, s));
```

Comment:
Don't use a static list, it can cause all sorts of weird behavior when there are multiple threads running this method.

Reply:
Revert it back